### PR TITLE
Remove DCOM support from *-Computer cmdlets (replacement for 5152)

### PR DESCRIPTION
**Pull request by JamesWTruher**
Since DCOM is not supported in corefx there was a great deal of dead code in the computer cmdlets.
This PR removes all vestiges of DCOM support from:

- Rename-Computer
- Restart-Computer
- Stop-Computer

removing about 4500 lines of dead code. Also, tests are updated to provide more complete coverage.
I also removed test-connection completely to make way for @iSazonov upcoming PR to improve coverage in tests, I created some test hook code which will test the cmdlet code without actually calling the WMI method to restart/rename/stop the system.

this is a replacement for PR 515

### CODE REVIEW

File: src/System.Management.Automation/engine/Utils.cs [hunk 1]
```diff
@@ -1416,20 +1416,29 @@ public static class InternalTestHooks
...
-        public static void SetTestHook(string property, bool value)
+        public static void SetTestHook(string property, object value)
         {
             var fieldInfo = typeof(InternalTestHooks).GetField(property, BindingFlags.Static | BindingFlags.NonPublic);
             if (fieldInfo != null)
             {
                 fieldInfo.SetValue(null, value);
             }
         }
+
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/Utils.cs]: Please remove extra line.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Rename-Computer.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        }
+
+        It "Should not produce a reboot warning when renaming a computer with the reboot flag" {
+            Set-TesthookResult -testhookName $RenameResultName -value $defaultResultValue
+            $newname = "mynewname"
+            $result = Rename-Computer -ErrorAction Stop -ComputerName . -NewName "$newname" -Pass -WarningAction SilentlyContinue -WarningVariable WarnVar -Restart
+            $result.HasSucceeded | should be $true
+            $result.NewComputerName | should be $newname
+            $WarnVar | should BeNullOrEmpty
+
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Rename-Computer.Tests.ps1]: Please remove extra line.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Restart-Computer.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                {
+                    $ProgressPreference=$pPref
+                }
+                $duration.TotalSeconds | Should BeGreaterThan $timeout
+            }
+            finally
+            {
+                Disable-Testhook -testhookname TestWaitStopComputer
+            }
+
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Restart-Computer.Tests.ps1]: Please remove extra line.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Stop-Computer.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            }
+        }
+
+        Context "Stop-Computer Error Conditions" {
+            It "Should return the proper error when it occurs" {
+                Set-TesthookResult -testhookName $stopTesthookResultName -Value 0x300000
+                Stop-Computer -ErrorVariable StopError 2>$null
+                $StopError.Exception.Message | Should match 0x300000
+            }
+
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Stop-Computer.Tests.ps1]: Please remove extra line.

**Review by iSazonov — APPROVED** [review]
LGTM with minor comments.

**Review by adityapatwardhan — APPROVED** [review]
LGTM

File: src/System.Management.Automation/engine/Utils.cs (hunk 1 above)
Comment (JamesWTruher) [reply src/System.Management.Automation/engine/Utils.cs]: fixed

File: test/powershell/Modules/Microsoft.PowerShell.Management/Rename-Computer.Tests.ps1 (hunk 2 above)
Comment (JamesWTruher) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Rename-Computer.Tests.ps1]: fixed

File: test/powershell/Modules/Microsoft.PowerShell.Management/Restart-Computer.Tests.ps1 (hunk 3 above)
Comment (JamesWTruher) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Restart-Computer.Tests.ps1]: fixed

File: test/powershell/Modules/Microsoft.PowerShell.Management/Stop-Computer.Tests.ps1 (hunk 4 above)
Comment (JamesWTruher) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Stop-Computer.Tests.ps1]: fixed

**Pull request merged**